### PR TITLE
bugfix: discard args of uri when using proxy-write plugin.

### LINKED ISSUE #635: discard args of uri when using proxy-write plugin
# Issue description
我使用 proxy-write 插件时，将原来的 url: /old/path 重写到新的 url：  /new/path，但是我的请求是有参数的 /old/path?id=2，rewrite 之后参数就没有了，改成 post 请求才可以，请问这个应该如何解决呢？

# Environment

* apisix version (cmd: `apisix version`): 0.7
* OS: docker Debian Linux 9

# What's the actual result? (include assertion message & call stack if applicable)

# What's the expected result?
希望转发 get 请求时，保留原来的参数

### PR DESCRIPTION
**Pull request by shuaijinchao**
### Issues resolved

Fix #635

### CODE REVIEW

File: t/plugin/proxy-rewrite.t [hunk 1]
```diff
@@ -403,3 +403,53 @@ host: localhost
...
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 15: rewrite uri args
+--- request
+GET /hello?q=apisix&a=iresty HTTP/1.1
+--- response_body
```
Comment (moonming) [inline t/plugin/proxy-rewrite.t]: not check the uri is `plugin_proxy_rewrite_args`

File: t/plugin/proxy-rewrite.t [hunk 2]
```diff
@@ -403,3 +403,53 @@ host: localhost
...
+--- request
+GET /t
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 15: rewrite uri args
```
Comment (moonming) [inline t/plugin/proxy-rewrite.t]: need add test case for not args.

File: t/plugin/proxy-rewrite.t [hunk 3]
```diff
@@ -403,3 +403,53 @@ host: localhost
...
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "proxy-rewrite": {
+                                "uri": "/plugin_proxy_rewrite_args",
```
Comment (moonming) [inline t/plugin/proxy-rewrite.t]: do we need to support `/plugin_proxy_rewrite_args?foo=123` in uri?

File: lua/apisix/plugins/proxy-rewrite.lua [hunk 4]
```diff
@@ -70,6 +70,13 @@ function _M.rewrite(conf, ctx)
         end
     end
 
+    local upstream_uri = conf.uri or ctx.var.uri
+    if ctx.var.is_args == "?" or str_sub(ctx.var.request_uri, -1) == "?" then
```
Comment (moonming) [inline lua/apisix/plugins/proxy-rewrite.lua]: Is  there a test case for `str_sub(ctx.var.request_uri, -1) == "?"` ? I am not get the point.

Comment (moonming) [inline lua/apisix/plugins/proxy-rewrite.lua]: is there a test case for `str_sub(ctx.var.request_uri, -1) == "?"`? I am not get your point.

File: lua/apisix/plugins/proxy-rewrite.lua [hunk 5]
```diff
@@ -70,6 +70,13 @@ function _M.rewrite(conf, ctx)
...
+        ctx.var["upstream_uri"] = upstream_uri .. "?" .. (ctx.var.args or "")
```
Comment (membphis) [inline lua/apisix/plugins/proxy-rewrite.lua]: `ctx.var.upstream_uri`, another code style

File: t/plugin/proxy-rewrite.t (hunk 2 above)
Comment (moonming) [reply t/plugin/proxy-rewrite.t]: and we need to modify doc at the same time.

File: t/plugin/proxy-rewrite.t (hunk 3 above)
Comment (shuaijinchao) [reply t/plugin/proxy-rewrite.t]: not supported.

Comment (shuaijinchao) [reply t/plugin/proxy-rewrite.t]: I want to look at the feedback in the community first to determine if it needs to be developed.

File: lua/apisix/plugins/proxy-rewrite.lua (hunk 4 above)
Comment (shuaijinchao) [reply lua/apisix/plugins/proxy-rewrite.lua]: I want to get the trailing `?` symbol in this way, but this way is wrong and has been deleted.

Comment (shuaijinchao) [reply lua/apisix/plugins/proxy-rewrite.lua]: ditto.

File: lua/apisix/plugins/proxy-rewrite.lua (hunk 5 above)
Comment (shuaijinchao) [reply lua/apisix/plugins/proxy-rewrite.lua]: updated.

File: t/plugin/proxy-rewrite.t (hunk 1 above)
Comment (shuaijinchao) [reply t/plugin/proxy-rewrite.t]: test cases have been added.

File: t/plugin/proxy-rewrite.t (hunk 2 above)
Comment (shuaijinchao) [reply t/plugin/proxy-rewrite.t]: test cases have been added.

**Pull request merged**